operations/observability/Observability.md

```diff
@@ -0,0 +1,365 @@
+
+### Configuration file for the grok-exporter
+
+The configuration file for the grok-exporter parses the Goss test log files and creates metrics from them.
+grok-exporter version 3 configuration is used for this task and to match log expressions with regular expressions.
+
+The following is an example configuration passed to the grok-exporter to get metrics:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example_name
+  namespace: example_namespace
+data:
+  config.yml: |-
+    global:
+      config_version: 3
+    input:
+      type: file
+      paths: 
+      - /logs/goss_tests/*
+      fail_on_missing_logfile: false
+    grok_patterns:
+      - 'REGEX [regular].*expression'
+      - 'EXAMPLE [aA-zZ-].*'
+    metrics:
+      - type: gauge
+        name: example_metric
+        help: EXAMPLE METRIC HELP
+        match: '{"log_file": "%{REGEX:logfile}", "TEST": "%{EXAMPLE:ex}"}'
+        value: '{{`{{.logfile}}`}}'
+        labels:
+          example_test: '{{`{{.ex}}`}}'
+    server:
+     port: 9144
+```
+
+### Prometheus metrics and Grafana dashboard
+
+After the preceding steps the `goss_tests` metrics are seen in Prometheus when Goss tests are run. Using these metrics,
+Grafana dashboards are created to shows the Goss tests details visually.
+
+Goss test dashboard features:
+
+- The dashboard has a dropdown for products, suites, and tests.
+- By default, all products, suites, and tests are selected for overall Goss tests result.
+- The overall product result, total number of the products, products passed, products failed, and its execution time is seen.
+- Suite results and test results are seen from the dashboard.
+- Node-wise test results are seen with failed nodes, passed nodes, and the suite/test description.
+
+![Goss test dashboard](../../img/operations/GossTestsDashboard.png "Goss Test Dashboard")
+
+## Error dashboard
+
+Error dashboards provide key glance at-a-insights about what is broken and needs attention.
+
+The automatic generation of the desired Prometheus grok-exporter configuration is based upon passing in a set of regular expression patterns to detect.
+This includes, but is not limited to, regular expression patterns to match and generate metrics on phases of manual administrator steps, as well as phases of installation scripts and automation.
+
+### Error dashboards features
+
+- Automate Grafana dashboards with error and failure message during CSM upgrade and install.
+- Create dashboard for issues comparison across multiple dimensions and clusters.
+- Monitor ConMan logs from the PIT node.
+- Parse log files with grok-exporter.
+- Create the Trend, Error, and Alert frequency dashboard.
+
+### Error dashboards
+
+Dashboards tracking different types of errors during install/upgrade.
+
+- Frequency of message about known issues dashboard
+- CSM environments install progress dashboard
```

Comment:
I see that environments is plural here. This dashboard monitors multiple CSM environments?

Reply:
Sure, but that is still the one CSM environment. Saying "CSM environments" doesn't sound like we're talking about a single CSM system anymore. I'd suggest just removing the word environments

```suggestion
- CSM install progress dashboard
```